src/app/components/autocomplete/autocomplete.component.ts

```diff
@@ -0,0 +1,149 @@
+import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
+import {
+	AbstractControl,
+	ControlValueAccessor,
+	FormControl,
+	NG_VALIDATORS,
+	NG_VALUE_ACCESSOR,
+	ValidationErrors,
+	Validators,
+} from '@angular/forms';
+import { combineLatest, debounceTime, map, Observable, of, startWith, switchMap } from 'rxjs';
+import { ErrorHandlerService, NpmRegistryService } from 'src/app/services';
+
+@Component({
+	selector: 'app-autocomplete',
+	templateUrl: './autocomplete.component.html',
+	styleUrls: ['./autocomplete.component.scss'],
+	changeDetection: ChangeDetectionStrategy.OnPush,
+	providers: [
+		{
+			provide: NG_VALUE_ACCESSOR,
+			useExisting: forwardRef(() => AutocompleteComponent),
+			multi: true,
+		},
+		{
+			provide: NG_VALIDATORS,
+			useExisting: forwardRef(() => AutocompleteComponent),
+			multi: true,
+		},
+	],
+})
+export class AutocompleteComponent implements ControlValueAccessor, Validators {
+	/*
+    keep track of package names that has been already loaded and show them
+    as suggestion for the user
+  */
+	@Input() autocompletePackageNames: string[] | null = [];
+
+	/*
+    Pacakge names that are already loaded to prevent selecting same library
+    multiple times
+  */
+	private alreadyLoadedPackageName: string[] = [];
+
+	/* Observale that will display loaded packages from NPM */
+	readonly autocompleteOptions$!: Observable<string[]>;
+
+	/* Form control to allow user search packages  */
+	readonly addPackage: FormControl<string> = new FormControl('', {
+		validators: this.errorHandlerService.noDuplicatesValidator(this.alreadyLoadedPackageName),
```

Comment:
Does passing `alreadyLoadedPackageName` like this work? I'd assume that this function is ran once on init and wouldn't use any updated values

Reply:
I think we don't even use this validation.
The reason is because `this.autocompleteOptions$`  get data from `getAutocompleteOptions()` that already prevents displaying the loaded package in the Select. So it will never happen that a package, which is already loaded will be displayed in the select. In my opinion, `validators: this.errorHandlerService.noDuplicatesValidator(this.alreadyLoadedPackageName)` is not important since it is never executed

Reply:
okej, maybe it is not unimportant, since you can manually type `@angular/cli` and select it multiple times

Reply:
fixed in: [58f066d](https://github.com/bitovi/opensource-analytics-dashboard/pull/9/commits/58f066d6d909637af48f8e19c09fe741006181ba)